### PR TITLE
fix: AAP-31364 catch parsing errors

### PR DESCRIPTION
**Pull request by mkanoor**
fix: check for parsing errors

https://issues.redhat.com/browse/AAP-31364
    
Catch parsing errors when handling incoming data

### CODE REVIEW

**Comment by Alex-Izquierdo** [comment]
Can you rebase? includes non related commits

File: tests/integration/api/test_event_stream_basic.py [hunk 1]
```diff
@@ -82,3 +82,46 @@ def test_post_event_stream_with_basic_auth(
...
+        "eda_credential_id": obj["id"],
+        "organization_id": get_default_test_org().id,
+        "test_mode": True,
+    }
+    event_stream = create_event_stream(admin_client, data_in)
+    user_pass = f"{username}:{secret}"
+
+    auth_value = f"Basic {base64.b64encode(user_pass.encode()).decode()}"
+    data_bytes = '{"a": 1,'.encode()
+    content_type = "application/json"
```
Comment (bzwei) [inline tests/integration/api/test_event_stream_basic.py:116]: The fix also enhanced error handling for `application/x-www-form-urlencoded`, better to include a test parameter for this content_type.

**Comment by mkanoor** [comment]
@bzwei @Alex-Izquierdo @hsong-rh Please review

**Review by hsong-rh — APPROVED** [review]

**Review by Alex-Izquierdo — APPROVED** [review]

File: tests/integration/api/test_event_stream_basic.py (hunk 1 above)
Comment (mkanoor) [reply tests/integration/api/test_event_stream_basic.py]: @bzwei Done

**Pull request merged**